ui/appui-react/src/appui-react/statusfields/message-center/MessageCenterField.tsx

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Notification
+ */
+import * as React from "react";
+import {
+  Button,
+  NotificationMarker,
+  Popover,
+  Tabs,
+} from "@itwin/itwinui-react";
+
+import { SvgChat } from "@itwin/itwinui-icons-react";
+import { Icon } from "@itwin/core-react";
+import { UiFramework } from "../../UiFramework";
+import { OutputMessagePriority } from "@itwin/core-frontend";
+import { MessageCenterMessage } from "./MessageCenterMessage";
+import { MessageManager } from "../../messages/MessageManager";
+import { TitleBar } from "../../layout/footer/dialog/TitleBar";
+import type { MessageType } from "@itwin/core-react";
+import type { NotifyMessageDetailsType } from "../../messages/ReactNotifyMessageDetails";
+import "./MessageCenterField.scss";
+
+/**
+ * Type for Status state to satisfy NotificationMarker type checking
+ */
+type status = "primary" | "negative";
+
+/** Message Center Field React component.
+ * @public
+ */
+export function MessageCenterField() {
+  const [messages, setMessages] = React.useState(MessageManager.messages);
+  const [notify, setNotify] = React.useState(false);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [status, setStatus] = React.useState<status>("primary");
+
+  const indicatorRef = React.createRef<HTMLButtonElement>();
+  const title = UiFramework.translate("messageCenter.messages");
+
+  const handleOpenChange = (isOpenState: boolean) => {
+    setNotify(false);
+    setIsOpen(isOpenState);
+  };
+
+  const isProblemStatus = (message: NotifyMessageDetailsType): boolean => {
+    // See priority values in DgnPlatform defined in NotificationManager
+    return (
+      message.priority === OutputMessagePriority.Error ||
+      message.priority === OutputMessagePriority.Fatal
+    );
+  };
+
+  const determineStatus = () =>
+    messages.some((msg) => isProblemStatus(msg))
+      ? setStatus("negative")
+      : setStatus("primary");
+
+  React.useEffect(() => {
+    MessageManager.registerAnimateOutToElement(indicatorRef.current);
+
+    return MessageManager.onMessagesUpdatedEvent.addListener(() => {
+      MessageManager.messages.length > 0 ? setNotify(true) : setNotify(false);
+      setMessages(MessageManager.messages);
+      determineStatus();
```

Comment:
This has to be moved to a `useEffect` that updates when `messages` state changes. In `determineStatus` function you use `messages` state which is not guaranteed to be updated by the time the function starts executing (if you're not sure why, I can explain further).